### PR TITLE
#12832980: register newsletter users mailchimp

### PR DESCRIPTION
**Pull request by bolorundurowb**

### CODE REVIEW

File: utils/MailChimpUtil.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+const MailChimp = require('mailchimp-api-v3');
+const Config = require('./../config/Config');
+
+const mailChimp = new MailChimp(Config.mailChimp.apiKey);
+
+class MailChimpUtil {
+  static async addUserToList(user) {
+    const listId = 'bd885c9eb2';
```
Comment (Saint-lawal) [inline utils/MailChimpUtil.js]: I think we should make this `listId` an environment variable for in case there is a change.

Comment (bolorundurowb) [reply utils/MailChimpUtil.js]: Done @Saint-lawal

File: controllers/Auth.js [hunk 2]
```diff
@@ -90,6 +91,27 @@ class Auth {
         // persist new user
         await user.save();
 
+        if (body.subscribe) {
+          let subscription = await db.Subscription
+            .findOne({
+              email: existingUser.email,
```
Comment (Saint-lawal) [inline controllers/Auth.js]: Hey, I believe this should be `user` not `existingUser`.

File: controllers/Auth.js [hunk 3]
```diff
@@ -90,6 +91,27 @@ class Auth {
...
+            });
+
+          // if a subscription does not exist, create one
+          if (!subscription) {
+            const user = await db.User
+              .findById(req.user.id);
+            await MailChimpUtil.addUserToList(user);
+
+            subscription = new db.Subscription({
+              email: existingUser.email,
```
Comment (Saint-lawal) [inline controllers/Auth.js]: Same here, `user` not `existingUser`.

**Pull request merged**